src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -872,8 +872,11 @@ impl Step for LlvmBitcodeLinker {
     fn run(self, builder: &Builder<'_>) -> PathBuf {
         let bin_name = "llvm-bitcode-linker";
 
-        builder.ensure(compile::Std::new(self.compiler, self.compiler.host));
-        builder.ensure(compile::Rustc::new(self.compiler, self.target));
+        // If enabled, use ci-rustc and skip building the in-tree compiler.
+        if !builder.download_rustc() {
+            builder.ensure(compile::Std::new(self.compiler, self.compiler.host));
```

Comment:
This could be resolved within the `impl Step for Std` logic, but I couldn't find a simple and clean solution without requiring a refactor. I wanted to unblock the https://github.com/rust-lang/rust/pull/122709 PR in the simplest way possible without introducing complex changes to the std bootstrapping logic.